assets/prefabs/Pineapple/Pineapple.prefab

```diff
@@ -0,0 +1,10 @@
+{
+  "parent": "engine:iconItem",
+  "DisplayName": {
+    "name": "Pineapple"
+  },
+  "Item": {
+    "icon": "SimpleFarming:SimpleFarming#pineapple",
```

Comment:
You have `assets/textures/Pineapple.png` to match this item, that would be "SimpleFarming:Pineapple" - the `#` means the preceding name is an atlas texture with an entry named "pineapple"